rust-runtime/inlineable/src/cbor_errors.rs

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use aws_smithy_cbor::decode::DeserializeError;
+use aws_smithy_cbor::Decoder;
+use aws_smithy_runtime_api::http::Headers;
+use aws_smithy_types::error::metadata::{Builder as ErrorMetadataBuilder, ErrorMetadata};
+
+fn sanitize_error_code(error_code: &str) -> &str {
+    // Trim a trailing URL from the error code, beginning with a `:`
+    let error_code = match error_code.find(':') {
+        Some(idx) => &error_code[..idx],
+        None => error_code,
+    };
+
+    // Trim a prefixing namespace from the error code, beginning with a `#`
+    match error_code.find('#') {
+        Some(idx) => &error_code[idx + 1..],
```

Comment:
Won't this panic if the error code ends with `#`?

Reply:
No, it won't, yielding an empty string code for error code. This function `sanitize_error_code` is a copy-paste from [json_errors::sanitize_error_code](https://github.com/smithy-lang/smithy-rs/blob/1223f61e10c82b5b8172ad9e18cd6cd558203751/rust-runtime/inlineable/src/json_errors.rs#L18), and I admit it's not ideal to duplicate code, but this is in `inlineable`. I see two options:
- still duplicate the function like above (with a comment saying this has been copied from `json_errors` and even duplicate the unit test
- move `sanitize_error_code` to a Smithy runtime crate and make it `pub`, maybe in `aws-smithy-http`?

Reply:
It won't, instead, will yield an empty string for error code. This function `sanitize_error_code` is a copy-paste from [json_errors::sanitize_error_code](https://github.com/smithy-lang/smithy-rs/blob/1223f61e10c82b5b8172ad9e18cd6cd558203751/rust-runtime/inlineable/src/json_errors.rs#L18), and I admit it's not ideal to duplicate code, but this is in `inlineable`. I see two options:
- still duplicate the function like above (with a comment saying this has been copied from `json_errors`) and even duplicate the unit test
- move `sanitize_error_code` to a Smithy runtime crate and make it `pub`, maybe in `aws-smithy-http`?

Reply:
No need to move `sanitize_error_code` to `aws-smithy-http`. I think David's question was about the slice starting at `idx + 1`, when `idx == len() - 1`. Since that won't panic, the implementation is okay. The conversation can be resolved.

Reply:
c9886c0 added a comment to `cbor_errors::sanitize_error_code` for clarification